### PR TITLE
func_i1_801922DC 

### PR DESCRIPTION
**Pull request by Vulto**
Stoled from Sonic Rider xD

### CODE REVIEW

File: src/overlays/ovl_i1/DFEE30.c [hunk 1]
```diff
@@ -3,7 +3,25 @@
 
 #pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/DFEE30/func_i1_801920F0.s")
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/DFEE30/func_i1_801922DC.s")
+extern s32 D_i1_8019C0B8;
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/DFEE30.c:6]: Can you move this extern to the top of the file after the includes?

Comment (Vulto) [reply src/overlays/ovl_i1/DFEE30.c:6]: Sure!

**Comment by Vulto** [comment]
Adjusted extern to the top of the file after the includes.

File: src/overlays/ovl_i1/DFEE30.c [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+
 #pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/DFEE30/func_i1_801920F0.s")
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/DFEE30/func_i1_801922DC.s")
+void func_i1_801922DC(Boss* this) {
+    Boss* boss = gBosses;
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/DFEE30.c:12]: does this match? remember to remove the boss++ below
```suggestion
for (i = 0; i < 4; i++, boss++) {

**Review by sonicdcer — APPROVED** [review]

File: src/overlays/ovl_i1/DFEE30.c [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
+for (i = 0; i < 4; i++, boss++) {
+        if ((boss->obj.status != 0) && (boss->obj.id == OBJ_BOSS_319)) {
+            if (boss->obj.pos.z <= this->obj.pos.z) {
+                D_i1_8019C0B8 = (s32)this->obj.rot.x + 1;
+                this->obj.status = 0;
+                break;
+            } else {
+                break;
+            }
+        }
```
Comment (inspectredc) [inline src/overlays/ovl_i1/DFEE30.c:21]:
```suggestion
        if ((boss->obj.status != 0) && (boss->obj.id == OBJ_BOSS_319)) {
            if (boss->obj.pos.z <= this->obj.pos.z) {
                D_i1_8019C0B8 = (s32)this->obj.rot.x + 1;
                this->obj.status = 0;
            }
            break;
        }
```

Would this match?

**Review by inspectredc — APPROVED** [review]

**Pull request closed without merging**